packages/gatsby-plugin-catch-links/src/__tests__/catch-links.js

```diff
@@ -351,3 +358,82 @@ describe(`navigation is routed through gatsby if the destination href`, () => {
     withSearch.dispatchEvent(clickEvent)
   })
 })
+
+describe(`pathPrefix is handled if href with ${pathPrefix}/someSubPath triggeres navigate to ${pathPrefix}/someSubPath`, () => {
+  // We're going to manually set up the event listener here
+  let hrefHandler
+  let eventDestroyer
+
+  beforeAll(() => {
+    hrefHandler = jest.fn()
+    eventDestroyer = catchLinks.default(window, hrefHandler)
+  })
+
+  afterAll(() => {
+    global.__PATH_PREFIX__ = ``
+    eventDestroyer()
+  })
+
+  test(`with pathPrefix='${pathPrefix}'`, done => {
+    global.__PATH_PREFIX__ = pathPrefix
+    const withPathPrefix = document.createElement(`a`)
+    withPathPrefix.setAttribute(
+      `href`,
+      `${window.location.href}/someSubPath`
+    )
+    document.body.appendChild(withPathPrefix)
+
+    // create the click event we'll be using for testing
+    const clickEvent = new MouseEvent(`click`, {
+      bubbles: true,
+      cancelable: true,
+      view: window,
+    })
+
+    hrefHandler.mockImplementation((path) => {
+      getNavigate()(path)
+      expect(global.___navigate).toHaveBeenCalledWith(
+        `${pathPrefix}/someSubPath`,
```

Comment:
will this test actually fail if you revert changes to `src/catch-links.js`? gotta say it's kind of confusing one to read ;)

Reply:
Just the one of them that simulates a build with --prefix-paths. That's the current state: module works in default builds but not in builds active prefix. But I wanted to catch the case too if any change would break the non prefix processing. As I said, these two are more of an integration test. I just don't know why I can't `import "gatsby-link"` there.

And yes, it was confusing. I tend to leave otherones code untouched. And this was a straight combination of the existing tests and one of the gatsby-link tests. I did some refactor and added comments.